File: go-version/leetcode/LC3201/maximum_length.go
package LC3201

/**
 * @Author: cutejiuge cutejiuge@163.com
 * @Date: 2025/7/16 下午11:41
 * @FilePath: maximum_length
 * @Description:
 */

func maximumLength(nums []int) int {
	var ans int
	patterns := [][]int{{0, 0}, {0, 1}, {1, 0}, {1, 1}}
	for _, pattern := range patterns {
		cnt := 0
		for _, num := range nums {
			if num&1 == pattern[cnt&1] {
				cnt++
			}
		}
		ans = max(ans, cnt)
	}
	return ans
}

File: python-version/LC3201/maximum_length.py
from typing import List


class Solution:
    @staticmethod
    def maximumLength(nums: List[int]) -> int:
        res = 0
        patterns = [[0, 0], [0, 1], [1, 0], [1, 1]]
        for pattern in patterns:
            cnt = 0
            for num in nums:
                if num & 1 == pattern[cnt & 1]:
                    cnt += 1
            res = max(res, cnt)
        return res
